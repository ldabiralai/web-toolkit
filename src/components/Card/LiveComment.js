import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import Link from '../../elements/Link';
import PlayIcon from '../../elements/PlayIcon';
import { brandBase, coreLightMinus1 } from '../../colors';
import { medium, large, wide } from '../../breakpoints';
import { fontFamilies } from '../../typography';

const StyledPictureContainer = styled.div`
  position: relative;
  border-radius: 4px;
  overflow: hidden;
  text-decoration: none;
  line-height: 0;
`;

export const StyledImage = styled.img`
  max-width: 100%;
  ${({ alt }) =>
    !alt &&
    css`
      border-radius: 4px;
    `}
`;

export const StyledPlayIcon = styled(PlayIcon)`
  position: absolute;
  bottom: 15px;
  left: 20px;
`;

export const StyledTitle = styled.div`
  display: block;
  min-height: 90px;
  padding: 15px 20px 28px;
  font-weight: bold;
  line-height: 20px;
  font-size: 14px;
  background-color: ${brandBase};
  color: ${coreLightMinus1};
  font-family: ${fontFamilies.interUi};
`;

const StyledLiveComment = styled.div`
  box-sizing: border-box;
  display: block;
  max-width: 286px;
  border-radius: 4px;
  overflow: hidden;
  text-decoration: none;

  ${medium(css`
    max-width: 388px;

    ${StyledTitle} {
      min-height: 102px;
      padding: 19px 24px 36px;
    }
  `)}
  
  ${large(css`
    max-width: 342px;
    ${StyledTitle} {
      font-size: 16px;
      min-height: 91px;
      padding: 15px 20px 28px;
    }
  `)}
  
  ${wide(css`
    max-width: 308px;
    ${StyledTitle} {
      font-size: 16px;
      padding: 15px 20px 33px;
    }
  `)}
  
`;

const LiveComment = ({ title, link, pictureUrl, isVideo }) => (
  <Link href={link} target="_blank">
    <StyledLiveComment>
      <StyledPictureContainer>
        <StyledImage src={pictureUrl} alt={title} />
        {isVideo && <StyledPlayIcon height={60} />}
      </StyledPictureContainer>
      {title && <StyledTitle>{title}</StyledTitle>}
    </StyledLiveComment>
  </Link>
);

LiveComment.defaultProps = {
  title: '',
  isVideo: false,
};

LiveComment.propTypes = {
  link: PropTypes.string.isRequired,
  pictureUrl: PropTypes.string.isRequired,
  title: PropTypes.string,
  isVideo: PropTypes.bool,
};
export default LiveComment;
