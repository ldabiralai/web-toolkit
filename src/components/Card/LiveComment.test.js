import React from 'react';
import { shallow } from 'enzyme';
import LiveComment, { StyledTitle, StyledImage, StyledPlayIcon } from './LiveComment';

const cardDetails = {
  pictureUrl: 'https://i.eurosport.com/2019/04/21/2570541-53309530-2560-1440.jpg?w=800',
  title: 'Federer continues to climb the rankings behind injured Rafael Nadal',
  link: 'www.google.com',
  isVideo: false,
};

describe('LiveComment card', () => {
  it('matches snapshot with valid props', () => {
    expect(shallow(<LiveComment {...cardDetails} />)).toMatchSnapshot();
  });

  it('renders PlayIcon when with video playback', () => {
    const wrapper = shallow(<LiveComment {...cardDetails} isVideo />);
    expect(wrapper.find(StyledImage)).toHaveLength(1);
    expect(wrapper.find(StyledPlayIcon)).toHaveLength(1);
  });

  it('renders card image when title is not provided', () => {
    const props = {
      ...cardDetails,
    };
    delete props.title;
    const wrapper = shallow(<LiveComment {...props} />);
    expect(wrapper.find(StyledImage)).toHaveLength(1);
    expect(wrapper.find(StyledTitle)).toHaveLength(0);
  });
});
