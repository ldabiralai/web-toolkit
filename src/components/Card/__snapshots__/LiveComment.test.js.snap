// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveComment card matches snapshot with valid props 1`] = `
<Link
  href="www.google.com"
  linkComponent={[Function]}
  target="_blank"
>
  <Styled(div)>
    <Styled(div)>
      <Styled(img)
        alt="Federer continues to climb the rankings behind injured Rafael Nadal"
        src="https://i.eurosport.com/2019/04/21/2570541-53309530-2560-1440.jpg?w=800"
      />
    </Styled(div)>
    <Styled(div)>
      Federer continues to climb the rankings behind injured Rafael Nadal
    </Styled(div)>
  </Styled(div)>
</Link>
`;
